### PR TITLE
Fix modern GTest & enable tests by default

### PR DESCRIPTION
**Pull request by JamiKettunen**
Here's all I could find in terms of `cmake` and `pkgconfig` in [the Void Linux `gtest` package](https://voidlinux.org/packages/?arch=x86_64&q=gtest):
```
$ xbps-query -Rf gtest-devel | grep -E 'cmake|pkgconfig'
/usr/lib/cmake/GTest/GTestConfig.cmake
/usr/lib/cmake/GTest/GTestConfigVersion.cmake
/usr/lib/cmake/GTest/GTestTargets-none.cmake
/usr/lib/cmake/GTest/GTestTargets.cmake
/usr/lib/pkgconfig/gmock.pc
/usr/lib/pkgconfig/gmock_main.pc
/usr/lib/pkgconfig/gtest.pc
/usr/lib/pkgconfig/gtest_main.pc
```

Perhaps we could add some modern distros (like `debian-unstable`) to CI as well to catch issues like this sooner? I've also enabled the option for building tests by default as I think it simply makes sense; what's the point for including them if they're not being exercised (on various distributions with different configurations) :p

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 # Options
 
-option(ENABLE_TESTS "Enable all tests and checks" OFF)
```
Comment (tari01) [inline CMakeLists.txt:12]: @sunweaver: I thing we should revert this change - it is inconsistent with everything else in Ayatana. In all our sources we provide defaults for quick and easy builds - testing is optional (mostly for packagers and developers in mind).

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
...
+option(ENABLE_TESTS "Enable all tests and checks" ON)
```
Comment (tari01) [inline CMakeLists.txt:12]: @sunweaver: I think we should revert this change - it is inconsistent with everything else in Ayatana. In all our sources we provide defaults for quick and easy builds - testing is optional (mostly for packagers and developers in mind).

Comment (JamiKettunen) [reply CMakeLists.txt:12]: This was reverted with https://github.com/AyatanaIndicators/ayatana-ido/commit/f19a5f472d282de32ca270afc68a6da8d4edca61.
